pkg/reduce/pnf/pnf.go

```diff
@@ -206,6 +209,20 @@ forwardLoop:
 				flush = true
 			}
 
+			if pf.windower.Strategy() == window.Accumulator {
+				winKey := strings.Join(response.WriteMessage.Keys, dfv1.KeysDelimitter)
+				if win, ok := pf.lastSeenWindow[winKey]; ok {
+					if win.EndTime().Before(response.Window.EndTime()) {
+						if err := pf.handleEOFWindow(ctx, response.Window); err != nil {
+							return
+						}
+						pf.lastSeenWindow[winKey] = response.Window
+					}
+				} else {
+					pf.lastSeenWindow[winKey] = response.Window
+				}
+			}
```

Comment:
now that we return WM as Window.Endtime, this can be simplified.

Reply:
This is more of an optimisation to persist the gc event only when the window end time changes. Added a comment.